arch/arm/core/aarch32/cpu_idle.S

```diff
@@ -73,6 +73,12 @@ SECTION_FUNC(TEXT, arch_cpu_idle)
 #error Unknown ARM architecture
 #endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
 
+	/* Synchronization barriers before calling WFI
+	* DSB is recommented by architecture definitions
```

Comment:
s/recommented/recommended

After a quick review of the ARM docs, I would say this is "required," not just "recommended."

Reply:
Note that this is only about AARCH32; actually I had only Cortex-M in mind. I interpreted "should" as "recommended" not "required". But it does not really make a difference, since we enforce it here anyways :)